packages/valora-auth/src/index.ts

```diff
@@ -0,0 +1,27 @@
+import * as functions from 'firebase-functions'
+import {
+  MOONPAY_PUBLIC_KEY_PROD,
+  MOONPAY_PUBLIC_KEY_STAGING,
+  RAMP_PUBLIC_KEY_PROD,
+  RAMP_PUBLIC_KEY_STAGING,
+  TRANSAK_PUBLIC_KEY_PROD,
+  TRANSAK_PUBLIC_KEY_STAGING,
+} from './config'
+
+export const distributeValoraKeys = functions.https.onRequest((request, response) => {
```

Comment:
Just a thought: 
If we generate the whole URI from here instead of just sending the keys, and we add the webhook integration for providers before the CB earn campaign we can push the tx hash of cash ins to firebase, maybe the `registrations` object so they users see the provider they cashed in from in the feed.

The things we would need for this to happen before the campaign in ~2 weeks are:
- Being able to modify URIs without doing a release (via this cloud function)
- Being able to read the sender of specific tx hashes from firebase (I can take this, should be straight forward enough)
- Being able to receive webhooks when a user cashes in with some provider (I already started working on this, I think we should be able to do it in time, at least for some providers)

What do you think? cc @jeanregisser

Reply:
Without this, users from the CB earn campaign which sounds like an important one will still see most of their cash ins as coming from 'Unknown'

Reply:
Update: Not doing this because of security concerns